### DISCUSSION TITLE
Why does the tool execution not working with Process.hierarchical?

### DISCUSSION THREAD

**Discussion by jais001** [crewAIInc/crewAI#1291] [Q&A]
Hi,
I have a set of tools that perform different tasks. And i have different agents to handle these tools. My idea was to provide a prompt to the Manger agent and execute the tools based on that. But while using tools with hierarchical process, tools are not being executed. Can I know why this is happening? Thanks for the help in advance.

**Comment by N8sGit** [comment]
I noticed this too, although I didn't make the connection that it was related to Process.hierarchical . I was trying to use the filewritetool to write files to a directory. I could trigger it manually no problem, but the agents fumbled with it hopelessly and insisted on passing JSON as arguments to it. It felt impossible to debug because it was happening somewhere in the LLM's black box.
